nipoppy/workflows/pipeline_store/list.py

```diff
@@ -0,0 +1,100 @@
+"""Workflow for pipeline list command."""
+
+from nipoppy.config.pipeline import BasePipelineConfig
+from nipoppy.env import PROGRAM_NAME, LogColor, PipelineTypeEnum
+from nipoppy.layout import DatasetLayout
+from nipoppy.utils import load_json
+from nipoppy.workflows.base import BaseDatasetWorkflow
+
+
+class PipelineListWorkflow(BaseDatasetWorkflow):
+    """List pipelines and versions that have been installed into a dataset."""
+
+    def __init__(
+        self,
+        dpath_root,
+        fpath_layout=None,
+        verbose=False,
+        dry_run=False,
+    ):
+        super().__init__(
+            dpath_root,
+            name="pipeline_list",
+            fpath_layout=fpath_layout,
+            verbose=verbose,
+            dry_run=dry_run,
+            _skip_logfile=True,
+        )
+
+    def _get_pipeline_info_map(
+        self,
+    ) -> dict[PipelineTypeEnum, dict[str, list[str]]]:
+        pipeline_type_to_info_map = {}
+        for pipeline_type in PipelineTypeEnum:
+            pipeline_names_to_versions_map = {}
+            dpath_pipeline_bundles = (
+                self.layout.dpath_pipelines
+                / DatasetLayout.pipeline_type_to_dname_map[pipeline_type]
+            )
+            for fpath_config in sorted(
+                dpath_pipeline_bundles.glob(f"*/{self.layout.fname_pipeline_config}")
+            ):
+                try:
+                    pipeline_config = BasePipelineConfig(**load_json(fpath_config))
+                except Exception as exception:
+                    raise RuntimeError(
+                        f"Error when loading pipeline config at {fpath_config}"
+                        f": {exception}"
+                    )
+
+                if pipeline_config.NAME not in pipeline_names_to_versions_map:
+                    pipeline_names_to_versions_map[pipeline_config.NAME] = []
```

Comment:
🍒 Could be avoided by converting`pipeline_names_to_versions_map` to a `defaultdict(list)`